packages/dm-core-plugins/src/generic-list/GenericTablePlugin.tsx

```diff
@@ -0,0 +1,190 @@
+      delete items[key]
+      setItems({ ...items })
+    })
+  }
+  const addItem = (reference: string, type: string) => {
+    dmssAPI
+      // TODO: Get type from parent blueprint, be able to select specialised type
+      .instantiateEntity({
+        entity: { type: items[Object.keys(items)[0]].type },
+      })
+      .then((newEntity: AxiosResponse<object, TGenericObject>) => {
+        dmssAPI
+          .documentAdd({ absoluteRef: reference, body: newEntity.data })
+          .then(() =>
+            setItems({
+              ...items,
+              [crypto.randomUUID()]: { ...newEntity.data },
+            })
+          )
+      })
+  }
+
+  const updateItem = (
+    key: string,
+    attribute: string,
+    newValue: string | number | boolean
+  ) => {
+    items[key][attribute] = newValue
+    setItems({ ...items })
+  }
+
+  const saveTable = () => {
+    setIsSaveLoading(true)
+    dmssAPI
+      .documentUpdate({
+        idReference: idReference,
+        data: JSON.stringify(Object.values(items)),
+      })
+      .catch((error: AxiosError<ErrorResponse>) =>
+        alert(JSON.stringify(error.response?.data))
+      )
+      .finally(() => setIsSaveLoading(false))
+  }
+
+  if (loading) return <Loading />
+  if (error) throw new Error(JSON.stringify(error, null, 2))
+  if (!items) return <Loading />
+
+  return (
+    <>
+      <div
+        style={{ display: 'flex', flexDirection: 'row-reverse', width: '100%' }}
+      >
+        <Button
+          variant="ghost_icon"
+          onClick={() => addItem(idReference, 'type')}
+        >
+          <Icon data={library_add} title="Append" />
+        </Button>
+      </div>
+      <Table style={{ width: '100%' }}>
+        <Table.Head>
+          <Table.Row>
+            {internalConfig.columns.map((attribute: string) => (
+              <Table.Cell key={attribute}>{attribute}</Table.Cell>
+            ))}
+            {internalConfig.editMode && internalConfig.showDelete && (
+              <Table.Cell>Delete</Table.Cell>
+            )}
+          </Table.Row>
+        </Table.Head>
+        <Table.Body>
+          {Object.entries(items).map(([key, item], index) => (
+            <Table.Row key={key}>
+              {internalConfig.columns.map((attribute: string) => {
+                if (typeof item[attribute] === 'object')
+                  throw new Error(
+                    `Objects can not be displayed in table header. Attribute '${attribute}' is not a primitive type.`
+                  )
+                // TODO: Consider having a more robust way of getting type
+                const attributeType = typeof item[attribute]
+                return (
+                  <Table.Cell key={attribute}>
+                    <Input
+                      value={item[attribute] || ''}
+                      readOnly={
+                        attribute === 'type' || !internalConfig.editMode
+                      }
+                      type={attributeType}
```

Comment:
I would argue that <Input type="number"> should not be used, and that <Input type="string"> is better due to validation

Reply:
This requires extra logic for converting data to correct types, and validating the entire form.
Added TODO that can be expanded on later.